packages/push/examples/smallest/main.rs

```diff
@@ -0,0 +1,215 @@
+pub mod args;
+
+use std::ops::Not;
+
+use anyhow::{ensure, Result};
+use clap::Parser;
+use ec_core::{
+    distributions::{collection::ConvertToCollectionGenerator, conversion::IntoDistribution},
+    generation::Generation,
+    individual::{ec::WithScorer, scorer::FnScorer},
+    operator::{
+        genome_extractor::GenomeExtractor,
+        genome_scorer::GenomeScorer,
+        mutator::Mutate,
+        selector::{best::Best, lexicase::Lexicase, Select, Selector},
+        Composable,
+    },
+    test_results::{self, TestResults},
+};
+use ec_linear::mutator::umad::Umad;
+use push::{
+    evaluation::cases::{Case, Cases, WithTargetFn},
+    genome::plushy::{GeneGenerator, Plushy},
+    instruction::{
+        variable_name::VariableName, BoolInstruction, ExecInstruction, IntInstruction,
+        PushInstruction,
+    },
+    push_vm::{program::PushProgram, push_state::PushState, stack::StackError, HasStack, State},
+};
+use rand::{distributions::Distribution, rngs::ThreadRng, thread_rng, RngCore};
+use strum::IntoEnumIterator;
+
+use crate::args::{Args, RunModel};
+
+// An input for this problem is a tuple of four `i64`s.
+#[derive(Copy, Clone)]
+struct Input(i64, i64, i64, i64);
+// An output for this problem is an `i64`.
+struct Output(i64);
+
+// This is an implementation of the "smallest" problem from Tom Helmuth's
+// software synthesis benchmark suite (PSB1):
+//
+// T. Helmuth and L. Spector. General Program Synthesis Benchmark Suite. In
+// GECCO '15: Proceedings of the 17th annual conference on Genetic and
+// evolutionary computation. July 2015. ACM.
+//
+// Problem Source: C. Le Goues et al., "The ManyBugs and IntroClass
+// Benchmarks for Automated Repair of C Programs," in IEEE Transactions on
+// Software Engineering, vol. 41, no. 12, pp. 1236-1256, Dec. 1 2015.
+// doi: 10.1109/TSE.2015.2454513
+//
+// This problem is quite easy if you have a `Min` instruction, but can be
+// more a bit more difficult without that instruction.
+fn main() -> Result<()> {
+    let mut rng = thread_rng();
+
+    let args = Args::parse();
```

Comment:
In the other examples I de-structure the args - might be preferable here as well